pkg/oam/workload.go

```diff
@@ -65,23 +66,44 @@ func BaseComplete(envName string, workloadName string, appGroup string, flagSet
 	}
 
 	for _, v := range template.Parameters {
-		flagValue, _ := flagSet.GetString(v.Name)
 		// Cli can check required flag before make a request to backend, but API itself could not, so validate flags here
-		if v.Required && v.Name != "name" && flagValue == "" {
-			return app, fmt.Errorf("required flag(s) \"%s\" not set", v.Name)
+		flag := flagSet.Lookup(v.Name)
+		if v.Name == "name" {
+			continue
+		}
+		if flag == nil || flag.Value.String() == "" {
+			if v.Required {
+				return nil, fmt.Errorf("required flag(s) \"%s\" not set", v.Name)
+			}
+			continue
 		}
 		switch v.Type {
 		case cue.IntKind:
-			d, _ := strconv.ParseInt(flagValue, 10, 64)
-			workloadData[v.Name] = d
+			workloadData[v.Name], err = flagSet.GetInt64(v.Name)
 		case cue.StringKind:
-			workloadData[v.Name] = flagValue
+			workloadData[v.Name], err = flagSet.GetString(v.Name)
 		case cue.BoolKind:
-			d, _ := strconv.ParseBool(flagValue)
-			workloadData[v.Name] = d
+			workloadData[v.Name], err = flagSet.GetBool(v.Name)
 		case cue.NumberKind, cue.FloatKind:
-			d, _ := strconv.ParseFloat(flagValue, 64)
-			workloadData[v.Name] = d
+			workloadData[v.Name], err = flagSet.GetFloat64(v.Name)
+		}
+		if err != nil {
```

Comment:
If v.Type doesn't meet any case, the value `err` is not the one returned by `flagSet.Getxxx()`. This might cause confusions.

Reply:
we should make sure only these for type will be stored in template.Prameter